File: ios/LedgerFlow/Models/InventoryModels.swift
import Foundation

struct InventoryItem: Codable {
    let id: String
    let userId: String
    let name: String
    let description: String
    let quantity: Int
    let unitPrice: Double
    let unit: String // "pcs", "kg", "liters", etc.
    let category: String
    let sku: String?
    let imageUrl: String?
    let createdAt: Date
    let updatedAt: Date
    
    init(
        id: String = UUID().uuidString,
        userId: String,
        name: String,
        description: String,
        quantity: Int,
        unitPrice: Double,
        unit: String,
        category: String,
        sku: String? = nil,
        imageUrl: String? = nil,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.userId = userId
        self.name = name
        self.description = description
        self.quantity = quantity
        self.unitPrice = unitPrice
        self.unit = unit
        self.category = category
        self.sku = sku
        self.imageUrl = imageUrl
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
    
    func toDictionary() -> [String: Any] {
        var dict: [String: Any] = [
            "id": id,
            "userId": userId,
            "name": name,
            "description": description,
            "quantity": quantity,
            "unitPrice": unitPrice,
            "unit": unit,
            "category": category,
            "createdAt": createdAt,
            "updatedAt": updatedAt
        ]
        if let sku = sku {
            dict["sku"] = sku
        }
        if let url = imageUrl {
            dict["imageUrl"] = url
        }
        return dict
    }
}

struct InventoryTransaction: Codable {
    let id: String
    let inventoryItemId: String
    let type: String // "in" or "out"
    let quantity: Int
    let reason: String // "purchase", "sale", "damage", "return"
    let reference: String?
    let date: Date
    let createdAt: Date
    
    init(
        id: String = UUID().uuidString,
        inventoryItemId: String,
        type: String,
        quantity: Int,
        reason: String,
        reference: String? = nil,
        date: Date = Date(),
        createdAt: Date = Date()
    ) {
        self.id = id
        self.inventoryItemId = inventoryItemId
        self.type = type
        self.quantity = quantity
        self.reason = reason
        self.reference = reference
        self.date = date
        self.createdAt = createdAt
    }
    
    func toDictionary() -> [String: Any] {
        var dict: [String: Any] = [
            "id": id,
            "inventoryItemId": inventoryItemId,
            "type": type,
            "quantity": quantity,
            "reason": reason,
            "date": date,
            "createdAt": createdAt
        ]
        if let ref = reference {
            dict["reference"] = ref
        }
        return dict
    }
}

File: android/app/src/main/kotlin/com/ledgerflow/models/InventoryModels.kt
package com.ledgerflow.models

import java.io.Serializable
import java.util.*

data class InventoryItem(
    val id: String = UUID.randomUUID().toString(),
    val userId: String,
    val name: String,
    val description: String,
    val quantity: Int,
    val unitPrice: Double,
    val unit: String, // "pcs", "kg", "liters", etc.
    val category: String,
    val sku: String? = null,
    val imageUrl: String? = null,
    val createdAt: Date = Date(),
    val updatedAt: Date = Date()
) : Serializable {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "userId" to userId,
            "name" to name,
            "description" to description,
            "quantity" to quantity,
            "unitPrice" to unitPrice,
            "unit" to unit,
            "category" to category,
            "sku" to sku,
            "imageUrl" to imageUrl,
            "createdAt" to createdAt,
            "updatedAt" to updatedAt
        )
    }
}

data class InventoryTransaction(
    val id: String = UUID.randomUUID().toString(),
    val inventoryItemId: String,
    val type: String, // "in" or "out"
    val quantity: Int,
    val reason: String, // "purchase", "sale", "damage", "return"
    val reference: String? = null,
    val date: Date = Date(),
    val createdAt: Date = Date()
) : Serializable {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "inventoryItemId" to inventoryItemId,
            "type" to type,
            "quantity" to quantity,
            "reason" to reason,
            "reference" to reference,
            "date" to date,
            "createdAt" to createdAt
        )
    }
}
